File: app/src/main/java/com/android/lowes/models/WeatherResponse.java
package com.android.lowes.models;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class WeatherResponse implements Parcelable {

    private Integer cnt;



    private List<WeatherDetails> list = null;

    public WeatherResponse() {
    }
    protected WeatherResponse(Parcel in) {
        if (in.readByte() == 0) {
            cnt = null;
        } else {
            cnt = in.readInt();
        }
    }

    public static final Creator<WeatherResponse> CREATOR = new Creator<WeatherResponse>() {
        @Override
        public WeatherResponse createFromParcel(Parcel in) {
            return new WeatherResponse(in);
        }

        @Override
        public WeatherResponse[] newArray(int size) {
            return new WeatherResponse[size];
        }
    };

    @Override
    public String toString() {
        return "Weather{" +
                "cnt=" + cnt +
                ", list=" + list +
                '}';
    }

    public Integer getCnt() {
        return cnt;
    }

    public void setCnt(Integer cnt) {
        this.cnt = cnt;
    }

    public List<WeatherDetails> getList() {
        return list;
    }

    public void setList(List<WeatherDetails> list) {
        this.list = list;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        if (cnt == null) {
            parcel.writeByte((byte) 0);
        } else {
            parcel.writeByte((byte) 1);
            parcel.writeInt(cnt);
        }
    }
}

File: app/src/main/java/com/android/lowes/models/WeatherResponse.kt
package com.android.lowes.models

import android.os.Parcel
import android.os.Parcelable

class WeatherResponse : Parcelable {
    var cnt: Int? = null
    var list: List<WeatherDetails>? = null

    constructor() {}
    protected constructor(`in`: Parcel) {
        cnt = if (`in`.readByte().toInt() == 0) {
            null
        } else {
            `in`.readInt()
        }
    }

    override fun toString(): String {
        return "Weather{" +
                "cnt=" + cnt +
                ", list=" + list +
                '}'
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        if (cnt == null) {
            parcel.writeByte(0.toByte())
        } else {
            parcel.writeByte(1.toByte())
            parcel.writeInt(cnt!!)
        }
    }

    companion object CREATOR : Parcelable.Creator<WeatherResponse> {
        override fun createFromParcel(parcel: Parcel): WeatherResponse {
            return WeatherResponse(parcel)
        }

        override fun newArray(size: Int): Array<WeatherResponse?> {
            return arrayOfNulls(size)
        }
    }
}